k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -1310,6 +1333,14 @@ public void setFlags(List<? extends Message> messages, final Set<Flag> flags, bo
             uids.add(Long.parseLong(message.getUid()));
         }
 
+        setFlagsInternal(uids, flags, value);
+    }
+
+    private void setFlagsInternal(Set<Long> uids, final Set<Flag> flags, boolean value)
+            throws MessagingException {
+        if (uids.isEmpty()) {
```

Comment:
Again, this should never happen. If it does we want to know about it, so throw an exception.